examples/fungible-token/src/storage_manager.rs

```diff
@@ -0,0 +1,94 @@
+use super::*;
+use near_sdk::json_types::{ValidAccountId, U128};
+use near_sdk::serde::Serialize;
+
+/// Price per 1 byte of storage from mainnet config after `0.18` release and protocol version `42`.
+/// It's 10 times lower than the genesis price.
+pub const STORAGE_PRICE_PER_BYTE: Balance = 10_000_000_000_000_000_000;
+
+#[derive(Serialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct AccountStorageBalance {
+    total: U128,
+    available: U128,
+}
+
+pub trait StorageManager {
+    fn storage_deposit(&mut self, account_id: Option<ValidAccountId>) -> AccountStorageBalance;
+
+    fn storage_withdraw(&mut self, amount: U128) -> AccountStorageBalance;
+
+    fn storage_minimum_balance(&self) -> U128;
+
+    fn storage_balance_of(&self, account_id: ValidAccountId) -> AccountStorageBalance;
+}
+
+#[near_bindgen]
+impl StorageManager for Contract {
+    #[payable]
+    fn storage_deposit(&mut self, account_id: Option<ValidAccountId>) -> AccountStorageBalance {
+        let amount = env::attached_deposit();
+        assert_eq!(
+            amount,
+            self.storage_minimum_balance().0,
+            "Requires attached deposit of the exact storage minimum balance"
+        );
+        let account_id = account_id
+            .map(|a| a.into())
+            .unwrap_or_else(|| env::predecessor_account_id());
+        if self.accounts.insert(&account_id, &0).is_some() {
+            env::panic(b"The account is already registered");
+        }
+        AccountStorageBalance {
+            total: amount.into(),
+            available: amount.into(),
+        }
+    }
+
+    #[payable]
+    fn storage_withdraw(&mut self, amount: U128) -> AccountStorageBalance {
```

Comment:
Something to call out here is that `storage_withdraw()` implementation here is specific to this FT use case. It is being used to "unregister" or "close" the account for this FT use case. Note that I say for "this" FT use case because different business rules may apply based on the specific FT use case, i.e., not all FT contracts may want this behavior.

In general, `storage_withdraw()` should only perform what it states, i.e., withdraw from the account's storage available balance. It should not be used to delete the account state from the contract. Closing the account should be performed explicitly via a separate function that makes it clear what the intent is.